src/tests/ftest/util/test_env_utils.py

```diff
@@ -0,0 +1,623 @@
+                if keys_found[provider_name] != hosts:
+                    keys_found.pop(provider_name)
+
+            # Select the preferred found provider based upon PROVIDER_KEYS order
+            log.debug("Supported providers detected: %s", list(keys_found))
+            for key in PROVIDER_KEYS:
+                if key in keys_found:
+                    provider = PROVIDER_KEYS[key]
+                    break
+
+        # Report an error if a provider cannot be found
+        if not provider:
+            raise TestEnvironmentException(
+                f"Error obtaining a supported provider for {self.interface} "
+                f"from: {list(PROVIDER_KEYS)}")
+
+        log.debug("  Found %s provider for %s", provider, self.interface)
+        return provider
+
+    @staticmethod
+    def __default_insecure_mode():
+        """Get the default insecure mode.
+
+        Returns:
+            str: the default insecure mode
+        """
+        return "True"
+
+    @staticmethod
+    def __default_bullseye_src():
+        """Get the default bullseye source file.
+
+        Returns:
+            str: the default bullseye source file
+        """
+        return os.path.join(os.path.dirname(os.path.abspath(__file__)), "test.cov")
+
+    @staticmethod
+    def __default_bullseye_file():
+        """Get the default bullseye file.
+
+        Returns:
+            str: the default bullseye file
+        """
+        return os.path.join(os.sep, "tmp", "test.cov")
+
+    def __set_environment(self):
+        """Set the test environment.
+
+        If a value is not set for the environment variable name set a default value
+        """
+        log = getLogger()
+        for value in self.__env_map.values():
+            if value['name'] not in os.environ:
+                log.debug('Setting the default value for %s', value['name'])
+                if value.get('kwargs', None):
+                    env_value = value['method'](**value['kwargs'])
+                else:
+                    env_value = value['method']()
+                if env_value is not None:
+                    os.environ[value['name']] = env_value
+
+    def set_with_hosts(self, servers, clients):
+        """Set the test environment with host information.
+
+        If a value does not exist for the interface and provider environment variable names it will
+        use the host information to determine a default value.
+
+        Args:
+            servers (NodeSet): hosts to be used in testing as a potential server
+            clients (NodeSet): hosts to be used in testing as a potential client
+
+        Raises:
+            TestEnvironmentException: if there is an error obtaining a default value to set
+        """
+        hosts = NodeSet()
+        hosts.update(servers)
+        hosts.update(clients)
+        self.__env_map['interface']['kwargs']['hosts'] = hosts
+        self.__env_map['provider']['kwargs']['hosts'] = servers
+        self.__set_environment()
```

Comment:
That these have to be handled manually here suggests to me that the "generic" approach in `__init__` just can't be generic. Especially since `__default_provider` and `__default_interface` are essentially the only `__default*` functions that have significant implementation.
Also, at this point, `__set_environment` would only update the interface and provider, so why not just update those two inline?